spec/features/users/sign_up_spec.rb

```diff
@@ -23,6 +23,21 @@
     end
   end
 
+  context 'user cancels on the enter password screen' do
+    it 'deletes the user record and returns them to the home page' do
+      email = 'test@test.com'
+
+      visit sign_up_email_path
+
+      submit_form_with_valid_email
+      click_confirmation_link_in_email(email)
+      click_on t('links.cancel_account_creation')
+
+      expect(current_path).to eq root_path
+      expect(User.find_with_email(email)).to be_nil
```

Comment:
Fair enough. This could be achieved with a view spec by making sure the link is making a DELETE call to the users controller.  @el-mapache The controller specs exist, but they did not catch this bug, hence the need for an additional test to make sure the link points to the right place.